pkg/controllers/termination/terminate.go

```diff
@@ -98,7 +99,7 @@ func (t *Terminator) drain(ctx context.Context, node *v1.Node) error {
 func (t *Terminator) terminate(ctx context.Context, node *v1.Node) error {
 	stored := node.DeepCopy()
 	// Delete the instance associated with node
-	if err := t.CloudProvider.Delete(ctx, node); err != nil {
+	if err := t.CloudProvider.Delete(ctx, v1alpha1.FromNode(node)); err != nil {
```

Comment:
Assuming this will eventually be a machine reconciler?

Reply:
Yes, the idea is that this will have to exist in parallel with a Machine termination finalizer so that we can handle the `kubectl delete node` case until `v1alpha6`. Still, the goal is that the CloudProvider interface should accurately reflect what the termination should look like on the Machine.

Reply:
For all things that can be removed in v1alpha6, can you add a comment that the method is "Deprecated", see: https://go.dev/blog/godoc?

Reply:
Sure, makes sense.

Reply:
I'm actually not convinced that any methods should be deprecated here yet. The termination flow isn't really deprecated and the NodeToMachine call could be utilized by the hydration controller.

Reply:
Both of those are deprecated by v1alpha6. The goal is to avoid additional dependency unless you know what you're doing.